### DISCUSSION TITLE
Dynamic RAM requirements for jobs to HTContor CEs

### DISCUSSION THREAD

**Discussion by AZSlow3** [DIRACGrid/DIRAC#6792] [Q&A] [answered]
I am aware of #6635, and we currently start to use mentioned approach. Unfortunately that does not solve our general problem.

The question has arisen in discussion between GridKa and LHCb. LHCb general RAM requirement is declared as  4GB per job (single core), but real mean RAM consumption is around 3GB (at the moment). While setting the max as fixed option for all queues is convenient for VO solution, that in a long term is not acceptable for resource providers since continuous overestimation leads to wasting resources and so farm efficiency.
That is not a problem as long as requested RAM is under physically available per core on nodes, which from what I know is the case for most VOs. But once requested RAM is higher, and 4GB per single core job is at the moment higher, efficient scheduling start to be problematic. If in addition extra asked amount is not used, resource provider has good reason to complain.

The question: is there general or HTCondor CE specific way to specify required for particular job RAM, so pilots are submitted with appropriate requirements?
Any other ideas how to solve the problem are also welcome.

**Comment by fstagni** [comment, marked as answer]
As discussed: the answer to the question is basically "no". 

This could anyway be turned into an issue where the PoolComputingElement would "partition" the node according to the available RAM.
